kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -145,37 +144,62 @@ void deferredRequestDelaysSubsequentRequest() {
         DecodedRequestFrame<?> outboundRequest1 = channel.readOutbound();
         assertThat(outboundRequest1).extracting(DecodedRequestFrame::body).isEqualTo(req1);
 
-        DecodedRequestFrame<?> outboundRequest2 = channel.readOutbound();
-        assertThat(outboundRequest2).extracting(DecodedRequestFrame::body).isEqualTo(req2);
+        OpaqueRequestFrame outboundRequest2 = channel.readOutbound();
+        assertThat(outboundRequest2).isSameAs(opaqueRequestFrame);
     }
 
     @Test
-    void deferredRequestDelaysSubsequentOpaqueRequest() {
+    void deferredRequestMethodsDispatchedOnEventloop() {
         var req1 = new ApiVersionsRequestData().setClientSoftwareName("req1");
+        var req2 = new ApiVersionsRequestData().setClientSoftwareName("req2");
 
         var requestFutureMap = new LinkedHashMap<ApiVersionsRequestData, CompletableFuture<Void>>();
         requestFutureMap.put(req1, new CompletableFuture<>());
+        requestFutureMap.put(req2, new CompletableFuture<>());
 
-        ApiVersionsRequestFilter filter = (apiVersion, header, request, context) -> requestFutureMap.get(request)
-                .thenCompose((u) -> context.forwardRequest(header, request));
+        // we expect the embedded netty eventloop to run on this thread
+        Thread expectedDispatchThread = Thread.currentThread();
```

Comment:
You've made an assumption about the behaviour of the netty mock.  That might change.  Elsewhere in the class, we've done the following to capture the thread.

```
 channel.eventLoop().submit(() -> eventLoopThreadFuture.complete(Thread.currentThread()));
channel.runPendingTasks();
assertThat(eventLoopThreadFuture).isCompleted();
```

we should extract that into a method and use it here too.